core/src/main/java/org/apache/brooklyn/feed/http/HttpFeed.java

```diff
@@ -275,7 +293,23 @@ public HttpFeed() {
     protected HttpFeed(Builder builder) {
         setConfig(ONLY_IF_SERVICE_UP, builder.onlyIfServiceUp);
         Map<String,String> baseHeaders = ImmutableMap.copyOf(checkNotNull(builder.headers, "headers"));
-        
+
+        HttpExecutor httpExecutor = null;
+        if (builder.httpExecutor != null) {
+            httpExecutor = builder.httpExecutor;
+        } else {
+            HttpExecutorFactory httpExecutorFactory = null;
+            Collection<Location> locations = builder.entity.getLocations();
+            Maybe<MachineLocation> location =  Machines.findUniqueMachineLocation(locations, MachineLocation.class);
```

Comment:
Use `Locations.getLocationsCheckingAncestors(locations, entity)` instead, the location could be coming from a parent entity.